cpp/arcticdb/storage/s3/detail-inl.hpp

```diff
@@ -125,6 +125,26 @@ namespace s3 {
                    || err == Aws::S3::S3Errors::RESOURCE_NOT_FOUND;
         }
 
+        inline void raise_if_unexpected_error(const Aws::S3::S3Error& err){
+            if (!is_expected_error_type(err.GetErrorType())) {
+                // We log a more detailed error explanation in case of NETWORK_CONNECTION errors to remedy #880.
+                if (err.GetErrorType() == Aws::S3::S3Errors::NETWORK_CONNECTION){
+                    log::storage().error("Got an unexpected network error: {}. "
+                                         "This could be due to a connectivity issue or exhausted open sockets. "
+                                         "In case socket file descriptors are exhausted, consider increasing `ulimit -n`.",
+                                         err.GetMessage().c_str());
+                }
+                else {
+                    log::storage().error("Got unexpected error: '{}' {}: {}",
+                                         int(err.GetErrorType()),
+                                         err.GetExceptionName().c_str(),
+                                         err.GetMessage().c_str());
+                }
+
+                throw UnexpectedS3ErrorException{};
```

Comment:
Does this bubble up to the Python layer? How does it look there?

I wonder if it is worth including more information in the exception itself, since in Jupyterhub our logging isn't shown properly.